### PR TITLE
crud info field column layout on sessions page

### PR DESCRIPTION
**Pull request by 31453**
rename everything LAYOUT instead of CONFIG

fixes [#2688](https://github.com/arkime/arkime/issues/2688)

## License
I confirm that this contribution is made under an Apache 2.0 license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by 31453** [comment]
Still need to display default arkime info field column layout and TESTSSSSSS

**Review by awick — CHANGES_REQUESTED** [review]
* CHANGELOG
* foo || [] should be foo ?? [] right?

File: viewer/apiUsers.js [hunk 1]
```diff
@@ -37,6 +37,12 @@ class UserAPIs {
     return user.columnConfigs || [];
   }
 
+  // --------------------------------------------------------------------------
+  static #userInfoFields (user) {
```
Comment (awick) [inline viewer/apiUsers.js:41]: this method only used in one place and couldn't it just be `user.?infoFieldsConfigs ?? []` :)

File: viewer/apiUsers.js [hunk 2]
```diff
@@ -87,6 +93,287 @@ class UserAPIs {
...
+   * @returns {object} user - The updated user object (if successful).
+   */
+  static #setSPIViewLayout (req) {
+    const result = UserAPIs.#validateSPIViewLayout(req);
+    if (!result.success) {
+      return result;
+    }
+
+    const user = req.settingUser;
+    user.spiviewFieldConfigs = user.spiviewFieldConfigs || [];
```
Comment (awick) [inline viewer/apiUsers.js:255]: user.spiviewFieldConfigs ??= []

File: viewer/apiUsers.js [hunk 3]
```diff
@@ -282,329 +578,149 @@ class UserAPIs {
...
-    for (let i = 0, len = user.spiviewFieldConfigs.length; i < len; i++) {
-      if (spiName === user.spiviewFieldConfigs[i].name) {
-        user.spiviewFieldConfigs[i] = req.body;
-        found = true;
-        break;
-      }
+  static updateUserLayout (req, res) {
+    let result;
+
+    if (req.params.type === 'sessionstable') {
```
Comment (awick) [inline viewer/apiUsers.js:657]: switch!

**Review by awick — DISMISSED** [review]

**Review by awick — APPROVED** [review]

**Pull request merged**